test.py

```diff
@@ -0,0 +1,191 @@
+    charsE = ''.join(sorted('CGKOSWaeimquy26-'))
+    # 0, 16, 32, 48: 0bxx0000
+    charsF = ''.join(sorted('AQgw'))
+    expected = [charsC, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsD, charsAll, charsE, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsF]
+    spreadTest(slugid.nice, expected)
+
+
+def testSpreadV4():
+    """ This test is the same as niceSpreadTest but for slugid.v4() rather than
+    slugid.nice(). The only difference is that a v4() slug can start with any of
+    the base64 characters since the first six bits of the uuid are random."""
+
+    charsAll = ''.join(sorted('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_'))
+    # 16, 17, 18, 19: 0b0100xx
+    charsD = ''.join(sorted('QRST'))
+    # 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62: 0bxxxx10
+    charsE = ''.join(sorted('CGKOSWaeimquy26-'))
+    # 0, 16, 32, 48: 0bxx0000
+    charsF = ''.join(sorted('AQgw'))
+    expected = [charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsD, charsAll, charsE, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsF]
+    spreadTest(slugid.v4, expected)
+
+
+def spreadTest(generator, expected):
+    """ `spreadTest` runs a test against the `generator` function, to check that
+    when calling it 64*40 times, the range of characters per string position it
+    returns matches the array `expected`, where each entry in `expected` is a
+    string of all possible characters that should appear in that position in the
+    string, at least once in the sample of 64*40 responses from the `generator`
+    function """
+    # k is an array which stores which characters were found at which
+    # positions. It has one entry per slugid character, therefore 22 entries.
+    # Each entry is a dict with a key for each character found, and its value
+    # as the number of times that character appeared at that position in the
+    # slugid in the large sample of slugids generated in this test.
+    k = [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
+
+    # Generate a large sample of slugids, and record what characters appeared
+    # where...  A monte-carlo test has demonstrated that with 64 * 20
+    # iterations, no failure occurred in 1000 simulations, so 64 * 40 should be
+    # suitably large to rule out false positives.
+    for i in range(0, 64 * 40):
+        slug = generator()
+        assert len(slug) == 22
+        for j in range(0, 22):
+            if slug[j] in k[j]:
+                k[j][slug[j]] = k[j][slug[j]] + 1
+            else:
+                k[j][slug[j]] = 1
+
+    # Compose results into an array `actual`, for comparison with `expected`
+    actual = []
+    for j in range(0, len(k)):
+        actual.append('')
+        for a in k[j].keys():
+            if k[j][a] > 0:
```

Comment:
iiuc, if `k[j][a] == 0` then `k[j][a]` will raise [KeyError](https://wiki.python.org/moin/KeyError), right? Doesn't it make more sense k be an array of [set](https://docs.python.org/2/library/stdtypes.html#set)s? So the test would be:

```python
if a in k[j]:
```